File: Unit2-04/Java/Dogs.java
/*
* This program uses a basic class to make an array of integers
*
* @author Michael Zagon
* @version 1.0
* @since 2024-04-01
*/

/**
* This is the program.
*/
final class Main {
    /**
    * Prevent instantiation.
    * Throw an exception IllegalStateException.
    * if this is ever called
    *
    * @throws IllegalStateException if this is ever called
    *
    */
    private Main() {
        throw new IllegalStateException("Cannot be instantiated");
    }

    /**
    * The starting main() function.
    *
    * @param args No args will be used
    */
    public static void main(final String[] args) {

        final MrCoxallStack dogStack = new MrCoxallStack();

        // Check size of stack
        System.out.println("The size of the stack is: " + dogStack.size());

        // Check boolean is empty
        System.out.println("Is the stack empty? " + dogStack.isEmpty());

        // Add fruit stack
        dogStack.push("poodle");
        dogStack.push("retriever");
        dogStack.push("beagle");
        System.out.println("Dog Items: " + dogStack.showStack());

        // Look at the top element of the list
        System.out.println("The top value is " + dogStack.peekItem());

        // Shows the popped item and new list
        final String topDog;
        topDog = dogStack.pop();
        System.out.println("Pop dog item: " + topDog);
        System.out.println("Dog Items: " + dogStack.showStack());

        // Check size of stack
        System.out.println("The size of the stack is: " + dogStack.size());

        // Check boolean is empty
        System.out.println("Is the stack empty? " + dogStack.isEmpty());

        // Clear the stack
        System.out.println("Emptying stack...");
        dogStack.clearStack();

        // Check size of stack
        System.out.println("The size of the stack is: " + dogStack.size());

        // Check boolean is empty
        System.out.println("Is the stack empty? " + dogStack.isEmpty());

        // Show the program as done
        System.out.println("\nDone.");
    }
}

File: Unit2-04/Typescript/dogs.ts
/**
* This program uses the MrCoxallStack
*
* By: Michael Zagon
* Version: 1.0
* Since:   2024-03-29
*/

import { MrCoxallStack } from './MrCoxallStack'

const dogStack = new MrCoxallStack()

// Checksize of stack
console.log("The size of the stack is: ", dogStack.size)

// Check boolean is empty
console.log("Is the stack empty? ", dogStack.isEmpty)

// Print fruit items
dogStack.push("poodle")
dogStack.push("retriever")
dogStack.push("beagle")
console.log("Dog items: ", dogStack.showStack())

// Look at the top element of the array
console.log("The top value is", dogStack.peekItem)

// Shows the popped item and new list
const topDog = dogStack.pop()
console.log("Pop dog item: ", topDog)
console.log("Dog items: ", dogStack.showStack())

// Checksize of stack
console.log("The size of the stack is: ", dogStack.size)

// Check boolean is empty
console.log("Is the stack empty? ", dogStack.isEmpty)

// Clear the stack
dogStack.clearStack()

// Checksize of stack
console.log("The size of the stack is: ", dogStack.size)

// Check boolean is empty
console.log("Is the stack empty? ", dogStack.isEmpty)

// Show the program as done
console.log('\nDone.')
